src/Symfony/Component/Validator/Constraints/RangeValidator.php

```diff
@@ -82,16 +87,37 @@ public function validate($value, Constraint $constraint)
                     throw new ConstraintDefinitionException(sprintf('The max value "%s" could not be converted to a "%s" instance in the "%s" constraint.', $max, $dateTimeClass, \get_class($constraint)));
                 }
             }
+        } elseif (($minIsDateIntervalComparison = DateIntervalComparisonHelper::supports($value, $min)) || ($maxIsDateIntervalComparison = DateIntervalComparisonHelper::supports($value, $max))) {
```

Comment:
the value of maxIsDateIntervalComparison depends on minIsDateIntervalComparison
shouldn't they be decoupled?

Reply:
It doesn't depend of it, I did this to avoid an useless call to `DateIntervalComparisonHelper::supports` but I just realized it's totally useless, it's gonna get called 2 times whatever happens 😅

Reply:
Actually it's good like that:
```php
if (($a = a()) || ($b = b()) {
    // $a is always set, $b potentially

    if ($a) {
        // ...
        
        // $a is true so $b is not set so:
        $b = $b();
    }

    if ($b) {
        // ...
    }
}
```

Reply:
`if (($a = a()) + ($b = b()) {`?

Reply:
or better:
```
else {
    if ($a = ...) { ...}
    if ($b = ...) { ...}
}
```

Reply:
In my example I forgot the fact that there is some processing done before the first "if", so something that must be done if `$a` or `$b` is true (the creation of the base `\DateTimeImmutable` to compare to and the conversion of the value in my case).

I think it's fine like that. I tried other ways but it's less readable to me.